docs/dev/table/sql/explain.md

```diff
@@ -0,0 +1,190 @@
+
+# explain SELECT statement through TableEnvironment.explain_sql()
+explanation1 = t_env.explain_sql(
+    "SELECT count, word FROM MyTable1 WHERE word LIKE 'F%' "
+    "UNION ALL "
+    "SELECT count, word FROM MyTable2")
+print(explanation1)
+
+# explain SELECT statement through TableEnvironment.execute_sql()
+table_result = t_env.execute_sql(
+    "EXPLAIN PLAN FOR "
+    "SELECT count, word FROM MyTable1 WHERE word LIKE 'F%' "
+    "UNION ALL "
+    "SELECT count, word FROM MyTable2")
+table_result.print()
+
+{% endhighlight %}
+</div>
+
+<div data-lang="SQL CLI" markdown="1">
+{% highlight sql %}
+Flink SQL> CREATE TABLE MyTable1 (count bigint, work VARCHAR(256);
+[INFO] Table has been created.
+
+Flink SQL> CREATE TABLE MyTable2 (count bigint, work VARCHAR(256);
+[INFO] Table has been created.
+
+Flink SQL> EXPLAIN PLAN FOR SELECT count, word FROM MyTable1 WHERE word LIKE 'F%' 
+> UNION ALL 
+> SELECT count, word FROM MyTable2;
+
+{% endhighlight %}
+</div>
+</div>
+
+The `EXPLAIN` result is:
+<div>
+{% highlight text %}
+== Abstract Syntax Tree ==
+LogicalUnion(all=[true])
+  LogicalFilter(condition=[LIKE($1, _UTF-16LE'F%')])
+    FlinkLogicalTableSourceScan(table=[[default_catalog, default_database, MyTable1]], fields=[count, word])
+  FlinkLogicalTableSourceScan(table=[[default_catalog, default_database, MyTable2]], fields=[count, word])
+  
+
+== Optimized Logical Plan ==
+DataStreamUnion(all=[true], union all=[count, word])
+  DataStreamCalc(select=[count, word], where=[LIKE(word, _UTF-16LE'F%')])
+    TableSourceScan(table=[[default_catalog, default_database, MyTable1]], fields=[count, word])
+  TableSourceScan(table=[[default_catalog, default_database, MyTable2]], fields=[count, word])
+
+== Physical Execution Plan ==
+Stage 1 : Data Source
+	content : collect elements with CollectionInputFormat
+
+Stage 2 : Data Source
+	content : collect elements with CollectionInputFormat
+
+	Stage 3 : Operator
+		content : from: (count, word)
+		ship_strategy : REBALANCE
+
+		Stage 4 : Operator
+			content : where: (LIKE(word, _UTF-16LE'F%')), select: (count, word)
+			ship_strategy : FORWARD
+
+			Stage 5 : Operator
+				content : from: (count, word)
+				ship_strategy : REBALANCE
+{% endhighlight %}
+</div>
+
+Similar to `EXPLAIN PLAN FOR SELECT ...`, we can also use `EXPLAIN PLAN FOR INSERT ...` statement to get the plan for INSERT statement. 
+
+{% top %}
+
+## Syntax
+
+{% highlight sql %}
+EXPLAIN PLAN 
+[EXCLUDING ATTRIBUTES | INCLUDING ALL ATTRIBUTES] 
+[WITH TYPE | WITH IMPLEMENTATION | WITHOUT IMPLEMENTATION]
+[AS XML | AS JSON] 
+FOR 
+<select_statement_or_insert_statement>
+
+{% endhighlight %}
+
+For select syntax, please refer to [SELECT]({{ site.baseurl }}/dev/table/sql/queries.html#supported-syntax) page.
+For insert syntax, please refer to [INSERT]({{ site.baseurl }}/dev/table/sql/insert.html) page.
+
+Currently, only `EXPLAIN PLAN FOR <select_or_insert_statement>` is supported.
```

Comment:
I would prefer not list the unsupported syntax (`EXCLUDING ATTRIBUTES`, etc...), because it doesn't help anything for users, but confuse users. Then we can remove the above line.